wagtail/api/v2/pagination.py

```diff
@@ -33,7 +36,8 @@ def paginate_queryset(self, queryset, request, view=None):
         stop = offset + limit
 
         self.view = view
-        self.total_count = queryset.count()
+        self.total_count = (queryset.count() if isinstance(queryset, QuerySet) or
+                            isinstance(queryset, BaseSearchResults) else len(queryset))
```

Comment:
`isinstance` can take a tuple of classes as its second argument, so you can write instead:
```python
isinstance(queryset, (QuerySet, BaseSearchResults))
```